src/govuk/components/pagination/pagination.yaml

```diff
@@ -0,0 +1,243 @@
+params:
+- name: previous
+  type: object
+  required: false
+  description: A link to the previous page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The link text to the previous page. Defaults to 'Previous page', where 'page' is visually hidden.
+  - name: visuallyHiddenText
+    type: string
+    required: false
+    description: The visually hidden label for the pagination item which will be applied to an `aria-label` and announced by screen readers on the previous page link.
+  - name: labelText
+    type: string
+    required: false
+    description: Optional label that goes underneath the previous page's link, providing further context for the user about where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The previous page's URL.
+  - name: attributes
+    type: object
+    required: false
+    description: HTML attributes (for example data attributes) to add to the anchor.
+- name: next
+  type: object
+  required: false
+  description: A link to the next page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The link text to the next page. Defaults to 'Next page', where 'page' is visually hidden.
+  - name: visuallyHiddenText
+    type: string
+    required: false
+    description: The visually hidden label for the pagination item which will be applied to an `aria-label` and announced by screen readers on the next page link.
+  - name: labelText
+    type: string
+    required: false
+    description: Optional label that goes underneath the next page link providing further context for where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The next page URL.
+  - name: attributes
+    type: object
+    required: false
+    description: HTML attributes (for example data attributes) to add to the anchor.
+- name: landmarkLabel
+  type: string
+  required: false
+  description: The label for the navigation landmark that wraps the pagination. Defaults to "results".
+- name: items
+  type: array
+  required: false
+  description: Array of link objects.
```

Comment:
I had a look at other arrays in govuk-frontend and we're kinda inconsistent. For example for the accordion tech docs, we add very little elaboration for the `items` array option:

```
An array of sections within the accordion.
```

However we elaborate much more in places like the cookie banner option `messages`:

```
The different messages you can pass into the cookie banner. For example, the cookie message and the confirmation message.
```

In this case I personally don't think we need to add an example because we're not doing anything unusual with this option. We're just assembling the pagination numbers, this feels quite easy to grasp to me.